services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.tsx

```diff
@@ -15,25 +15,111 @@ along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>
+import { RolloutStatus, StreamStatusResponse } from '../../../api/api';
 
 export type ReleaseCardProps = {
     className?: string;
     version: number;
     app: string;
 };
 
+const RolloutStatusIcon: React.FC<{ status: RolloutStatus }> = (props) => {
+    const { status } = props;
+    switch (status) {
+        case RolloutStatus.RolloutStatusSuccesful:
+            return <span className="rollout__icon_successful">✓</span>;
+        case RolloutStatus.RolloutStatusProgressing:
+            return <span className="rollout__icon_progressing">↻</span>;
+        case RolloutStatus.RolloutStatusError:
+            return <span className="rollout__icon_error">!</span>;
+    }
+    return <span className="rollout__icon_unknown">?</span>;
+};
+
+const RolloutStatusDescription: React.FC<{ status: RolloutStatus }> = (props) => {
+    const { status } = props;
+    switch (status) {
+        case RolloutStatus.RolloutStatusSuccesful:
+            return <span className="rollout__description_successful">✓ Done</span>;
+        case RolloutStatus.RolloutStatusProgressing:
+            return <span className="rollout__description_progressing">↻ In progress</span>;
+        case RolloutStatus.RolloutStatusError:
+            return <span className="rollout__description_error">! Failed</span>;
+    }
+    return <span className="rollout__description_unknown">? Unkwown</span>;
+};
+
+const rolloutStatusPriority = [
+    RolloutStatus.RolloutStatusError,
+    RolloutStatus.RolloutStatusProgressing,
+    RolloutStatus.RolloutStatusUnknown,
+];
+
+const calculateDeploymentStatus = (
+    app: string,
+    deployedAt: EnvironmentGroupExtended[],
+    rolloutStatus: RolloutStatusApplication
+): [Array<StreamStatusResponse>, StreamStatusResponse?] => {
+    const rolloutEnvs = deployedAt.flatMap((envGroup) =>
+        envGroup.environments.map((env) => {
+            const status = rolloutStatus[env.name];
+            if (!status) {
+                // The status is completly unknown. Either the app is just new or rollout service is not responding.
+                return {
+                    environment: env.name,
+                    rolloutStatus: RolloutStatus.RolloutStatusUnknown,
+                    version: 0,
+                    application: app,
+                };
+            }
+            if (
+                status.rolloutStatus === RolloutStatus.RolloutStatusSuccesful &&
+                status.version !== env.applications[app]?.version
+            ) {
+                // The rollout service might be sligthly behind the UI.
+                // In that case the
+                return { ...status, rolloutStatus: RolloutStatus.RolloutStatusProgressing };
+            }
+            return status;
+        })
+    );
+    rolloutEnvs.sort((a, b) => {
+        if (a.environment < b.environment) {
+            return -1;
+        } else if (a.environment > b.environment) {
+            return 1;
+        }
+        return 0;
```

Comment:
fine for now, but later on we should group by envGroup as well

Reply:
Maybe yes, on the other hand, it's also interesting to see it for each cluster. What do you think we should do?

Reply:
yes, show for each cluster, but sort/group-by env group.